rust/src/settings_manager.rs

```diff
@@ -1,72 +1,133 @@
-use std::sync::Mutex;
+use std::fmt;
 
-use yaml_rust::{Yaml, YamlLoader};
+use serde::{
+    de::{self, Unexpected},
+    Deserialize,
+};
 
-const SOURCE: &str = include_str!("../../settings.yaml");
+lazy_static::lazy_static! {
+    static ref SETTINGS: Vec<Setting> = {
```

Comment:
Originally the settings were wrapped in a mutex, which we don't need if this is going to be readonly (which I think it is)

Reply:
👍